### ISSUE TITLE
Validate AccountEmail parameter for account requests

### ISSUE THREAD

**Issue by bmurphey** [aws-ia/terraform-aws-control_tower_account_factory#293]
**Describe the outcome you'd like**

I would like the AccountEmail parameter in new account requests to be validated before being applied.

**Is your feature request related to a problem you are currently experiencing? If so, please describe.**

The max length for AccountEmail is 64 characters, but the `ct-aft-account-request` pipeline will run successfully even if the email is too long.  This will cause an error downstream in Service Catalog when provisioning.

**Additional context**

Validation similar to this will fail the `ct-aft-account-request` pipeline before the Service Catalog provisioning is attempted:
```
variable "control_tower_parameters" {
  description = "Parameters from Control Tower to pass in."
  type = object({
    AccountEmail              = string
    AccountName               = string
    ManagedOrganizationalUnit = string
    SSOUserEmail              = string
    SSOUserFirstName          = string
    SSOUserLastName           = string
  })
  validation {
    condition     = length(var.control_tower_parameters.AccountEmail) < 65
    error_message = "AccountEmail parameter must be 64 characters or less"
  }
}
```

**Comment by adam-daily** [comment]
Hey Brian, thanks for the request, I'll bring this to the team to have a look.

**Comment by pmmalinov01** [comment]
We have written OPA test we run on the Pr to check if we are using an alias the alias is the account name and it matches our domain and is a proper email address. may you can follow the same approach

**Comment by arongate** [comment]
We need this! We need to implements some custom validations for account request. Can you please add some account request validation framework?

**Comment by pmmalinov01** [comment]
> We need this! We need to implements some custom validations for account request. Can you please add some kind of  account request validation framework?

You can add tests your self, there are a lot of options out there

**Comment by arongate** [comment]
> > We need this! We need to implements some custom validations for account request. Can you please add some kind of  account request validation framework?
> 
> You can add tests your self, there are a lot of options out there

No, I can't. In my case I need it to be added to the account factory. 
There is two separated roles/teams. One for aft deployment/management and another for aft customizations repositories and CI.
Choosing another option is not an option.

**Comment by eriweb** [comment]
This is rather simple to implement yourself, but for newcomers this might be something they know nothing about and cleaning up failed resources is both annoying and unnecessary in this instance as it's easily avoidable. 

I really think AWS should add these constraints to the example as it would easily make adoption of AFT feel better.

For any interested, this is one way to do it:

```
variable "control_tower_parameters" {
  type = object({
    AccountEmail              = string
    AccountName               = string
    ManagedOrganizationalUnit = string
    SSOUserEmail              = string
    SSOUserFirstName          = string
    SSOUserLastName           = string
  })

  validation {
    condition     = length(var.control_tower_parameters.AccountEmail) <= 64
    error_message = "AccountEmail must not be longer than 64 characters."
  }

  validation {
    condition     = length(var.control_tower_parameters.AccountName) <= 50
    error_message = "AccountName must not be longer than 50 characters."
  }
}
```
